python/ray/ml/tests/test_lightgbm_trainer.py

```diff
@@ -0,0 +1,175 @@
+    valid_dataset = ray.data.from_pandas(test_df)
+    trainer = LightGBMTrainer(
+        scaling_config=scale_config,
+        label_column="target",
+        lightgbm_config={PARAMS_KEY: params, "num_boost_round": 5},
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
+    )
+    result = trainer.fit()
+    checkpoint = result.checkpoint
+    lgbm_model = load_model_from_checkpoint(checkpoint)
+    assert get_num_trees(lgbm_model) == 5
+
+    # Move checkpoint to a different directory.
+    checkpoint_dict = result.checkpoint.to_dict()
+    checkpoint = Checkpoint.from_dict(checkpoint_dict)
+    checkpoint_path = checkpoint.to_directory(tmpdir)
+    resume_from = Checkpoint.from_directory(checkpoint_path)
+
+    trainer = LightGBMTrainer(
+        scaling_config=scale_config,
+        label_column="target",
+        lightgbm_config={PARAMS_KEY: params, "num_boost_round": 5},
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
+        resume_from_checkpoint=resume_from,
+    )
+    result = trainer.fit()
+    checkpoint = result.checkpoint
+    lgbm_model = load_model_from_checkpoint(checkpoint)
+    assert get_num_trees(lgbm_model) == 10
+
+
+def test_preprocessor_in_checkpoint(ray_start_4_cpus):
+    train_dataset = ray.data.from_pandas(train_df)
+    valid_dataset = ray.data.from_pandas(test_df)
+
+    class DummyPreprocessor(Preprocessor):
+        def __init__(self):
+            super().__init__()
+            self.is_same = True
+
+        def fit(self, dataset):
+            self.fitted_ = True
+
+        def _transform_pandas(self, df: "pd.DataFrame") -> "pd.DataFrame":
+            return df
+
+    trainer = LightGBMTrainer(
+        scaling_config=scale_config,
+        label_column="target",
+        lightgbm_config={PARAMS_KEY: params},
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
+        preprocessor=DummyPreprocessor(),
+    )
+    result = trainer.fit()
+    lgbm_model = load_model_from_checkpoint(result.checkpoint)
+    assert get_num_trees(lgbm_model) == 10
+    assert result.checkpoint.to_dict()[PREPROCESSOR_KEY].is_same
+    assert result.checkpoint.to_dict()[PREPROCESSOR_KEY].fitted_
+
+
+def test_tune(ray_start_4_cpus):
+    train_dataset = ray.data.from_pandas(train_df)
+    valid_dataset = ray.data.from_pandas(test_df)
+    trainer = LightGBMTrainer(
+        scaling_config=scale_config,
+        label_column="target",
+        lightgbm_config={PARAMS_KEY: {**params, **{"max_depth": 1}}},
+        datasets={TRAIN_DATASET_KEY: train_dataset, "valid": valid_dataset},
+    )
+
+    tune.run(
+        trainer.as_trainable(),
+        config={"lightgbm_config": {PARAMS_KEY: {"max_depth": tune.randint(2, 4)}}},
+        num_samples=2,
+    )
+
+    # Make sure original Trainer is not affected.
+    assert trainer.lightgbm_config[PARAMS_KEY]["max_depth"] == 1
+
+
+def test_validation(ray_start_4_cpus):
```

Comment:
Can we also add a e2e test that includes loading the checkpoint into the predictor as well? Like what we have for TorchTrainer #23219

Reply:
This can just be loading into the predictor and doing inference on some dummy dataset.

Reply:
That will be covered in an example I am adding rn